### PR TITLE
Registration changed for Laravel 5.4

### LINKED ISSUE #4: Not working for Laravel 5.4
After upgrading to Laravel 5.4, running migrations results in the following error:
```
Call to undefined method Illuminate\Database\Schema\Blueprint::passthru()
```

From the [Upgrade Guide](https://laravel.com/docs/5.4/upgrade):

#### Custom Connections

If you were previously binding a service container binding for a `db.connection.{driver-name}` key in order to resolve a custom database connection instance, you should now use the  `Illuminate\Database\Connection::resolverFor` method in the `register` method of your `AppServiceProvider`:

```php
use Illuminate\Database\Connection;

Connection::resolverFor('driver-name', function ($connection, $database, $prefix, $config) {
    //
});
```

### PR DESCRIPTION
**Pull request by SebastianS90**
closes #4 

As this is a breaking change, I suggest to release it as version `2.0` and putting a note to the readme that `^1.0` is for Laravel `< 5.4` and `^2.0` is for Laravel `>= 5.4`.

### CODE REVIEW

**Comment by SebastianS90** [comment]
Not sure about the test problems. Travis [succeeded](https://travis-ci.org/shiftonelabs/laravel-nomad/jobs/195867459) for PHP 7 but [fails](https://travis-ci.org/shiftonelabs/laravel-nomad/builds/195867456) for the other environments. The reason might be my change from phpunit `~4.0` to `~5.0`. I had to update the dependency because of [this error](https://travis-ci.org/shiftonelabs/laravel-nomad/jobs/195841399#L280):
```
PHP Fatal error:  Class 'PHPUnit\Framework\TestCase' not found in /home/travis/build/shiftonelabs/laravel-nomad/vendor/laravel/framework/src/Illuminate/Foundation/Testing/TestCase.php on line 12
```

**Comment by SebastianS90** [comment]
I got the tests working on travis. The `phpunit` command just uses what is installed in the test VM, and that might not be compatible with what is pulled in by composer. So I switched to `./vendor/bin/phpunit`, thus ensuring compatibility.
Furthermore, Laravel and the current phpunit version do not support php 5.5 any more, so I removed it from the travis file.

In case that you release this backwards incompatible change as version `2.0.0`, then you can just use #6 to document how to use the correct version.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 language: php
 
 php:
-  - 5.5
```
Comment (patrickcarlohickman) [inline .travis.yml:4]: Keep support for 5.5.

File: composer.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
         "illuminate/support": ">=4.1"
     },
     "require-dev": {
-        "phpunit/phpunit": "~4.0",
```
Comment (patrickcarlohickman) [inline composer.json:23]: I believe the older version is required to keep support for L4.

File: src/LaravelNomadServiceProvider.php [hunk 3]
```diff
@@ -23,9 +28,17 @@ public function boot()
...
+        });
+        Connection::resolverFor('pgsql', function ($connection, $database, $prefix, $config) {
+            return new PostgresConnection($connection, $database, $prefix, $config);
+        });
+        Connection::resolverFor('sqlite', function ($connection, $database, $prefix, $config) {
+            return new SQLiteConnection($connection, $database, $prefix, $config);
+        });
+        Connection::resolverFor('sqlsrv', function ($connection, $database, $prefix, $config) {
+            return new SqlServerConnection($connection, $database, $prefix, $config);
+        });
```
Comment (patrickcarlohickman) [inline src/LaravelNomadServiceProvider.php:42]: Do a check on `method_exists(Connection::class, 'resolverFor')`. If the method exists, use this new registration method. If the method doesn't exist, keep the old bindings.

File: tests/TestCase.php [hunk 4]
```diff
@@ -3,8 +3,8 @@
 
 use ReflectionMethod;
 use Illuminate\Foundation\Application;
+use Illuminate\Database\Connectors\ConnectionFactory;
```
Comment (patrickcarlohickman) [inline tests/TestCase.php:6]: To be consistent with Laravel, I keep my `use` statements in ascending order of length. This needs to be moved down a couple lines.

**Review by patrickcarlohickman — CHANGES_REQUESTED** [review]
Changes to prevent new major version for L5.4 support.

**Comment by SebastianS90** [comment]
Closed in favor of #7

**Pull request closed without merging**